### DISCUSSION TITLE
Lesson 4: Got ExtraData error and the fix

### DISCUSSION THREAD

**Discussion by weilainicolas** [smartcontractkit/full-blockchain-solidity-course-py#524] [General]
When I tried to deploy the contract to rinkeby, I got the following error: web3.exceptions.ExtraDataLengthError: The field extraData is 97 bytes, but should be 32.

What I found online is that when we connect to POA network using Web3, it may cause this error. It is because POA will add additional data at extraData, which could exceed the required 32 bytes. 
Found a fix online and it worked. 

```Python

from web3 import Web3
from web3.middleware import geth_poa_middleware

w3 = Web3('http://localhost:8545')
w3.middleware_onion.inject(geth_poa_middleware, layer=0)
```

I hope this can help people who have the similar issues.

**Comment by cromewar** [comment]
excellent contribution to the community @weilainicolas I hope this info might help for other users.

**Reply by Cosmodude** [reply]
For clearance, where should this code be pasted?
